driftbase/models/db.py

```diff
@@ -114,7 +112,35 @@ def check_password(self, password):
 
 tbl_user_identity = UserIdentity.__table__
 
-
+class PlayerRichPresence:
+    """
+    Rich presence information for a particular user.
+    @see PlayerRichPresenceSchema
+    """
+    
+    game_mode = ""
+    map_name = ""
+    is_online = False
+    is_in_game = False
+
+    def __init__(self, player_id: int, is_online: bool):
+        self.is_online = is_online
+
+        try:
+            match_player : MatchPlayer = g.db.query(MatchPlayer) \
+                .filter(MatchPlayer.player_id == player_id) \
+                .one()
+            
+            match : Match = g.db.query(Match) \
+                .filter(Match.match_id == match_player.match_id) \
+                .one()
+
+            self.is_in_game = match_player.status == "active" # Always true?
+            self.game_mode = match.game_mode
+            self.map_name = match.map_name
+        except Exception: # Expected NoResultFound if a player isn't in a match.
```

Comment:
Please use a more specific exception, so that actual errors can be dealt with differently from not found